### DISCUSSION TITLE
Microchip Studio - new Project as Arduino Sketch Import - no megaTinyCore, no DxCore ?

### DISCUSSION THREAD

**Discussion by apws** [SpenceKonde/megaTinyCore#823] [Q&A]
Hi, I tried to switch chip/board on my test sketch from 328p Xplained Mini to CuriosityNano ATtiny1627 and, it seems it isnt that easy only to select device and rebuild as generated code isnt modified any way, but I was surprised also that when I try to make new project based on Arduino sketch, then your megaTinyCore and DxCore chips/boards are not listed in wizzard, only ATTinyCore is here and obviously all Arduinos and old Atmel Xplained Minis. Can I do something to do such conversion? Or is there some hack? As replacing include/vyriants/xxxx/pins_arduino.h ??

Thanks,
Petr

**Comment by apws** [comment]
hi, guys, is it too political or what? Is there some blocking thing why it is no supported? :-(

**Comment by ObviousInRetrospect** [comment]
I'm having a hard time understanding why on earth you would want to move a project that can be worked on outside of Microchip Studio into Microchip Studio. I have had to use that piece of ... software (not the first thing starting with s that comes to mind) for some DD work and hate every minute of it.

I suspect the reason you aren't getting a response is probably the lack of experience with what you are trying to do over here. You might ask over somewhere more Microchip Studio focused. From my perspective your question is about Microchip Studio's support for any non-default cores.

Out of curiosity, why do you want to use Microchip Studio? Is there something it does well for you that either Arduino (2.0, 1.8.19) or PlatformIO lacks? The only thing that comes to mind is its debugger. That I have used by importing a pre-built project.

**Reply by apws** [reply]
ya, debuger, I finally solved it by completely getting rid of arduino, or rather preprocessing #ifdefs to work anywhere, mocking part of C++ arduino access (so I build also in Dev-C++, where debugging is FASTEST ... 100x faster than anywhere else), ... for me AtmelStudio is okay, but I finally moved to even uglier MPLAB X, where I can compile code even for PIC ... but thats not all, I am now back in C#, hopefully forever :-)

**Reply by raghunathanmohan** [reply]
Hi Petr, I am Raghunathan. I am trying to do the same thing like you. I am able to build the SW in Arduino. Unfortunately there is a reset happening in the HW and I want to debug. Please provide more information on how to - preprocess #ifdefs so that it can be used anywhere. I also do not know which files from the magatinycore are needed in the microchip studio project. It would be of great help if you could also help me here.
